### PR TITLE
mes-7246: replaced confirm test submit modal

### PR DESCRIPTION
**Pull request by solomonsimukoko**
mes-7246 replace confirm screen alert

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

Screenshot 2021-09-29 at 15 11 27

### CODE REVIEW

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/__tests__/submit-modal.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    modalComponent = modalFixture.componentInstance;
+    // spyOn(modalComponent.store$, 'dispatch').and.callFake(() => {});
+    spyOn(modalComponent.modalController, 'dismiss').and.returnValue(Promise.resolve(true));
+  }));
+
+  it('should call onBack when the Cancel button is clicked', async () => {
+    spyOn(modalComponent, 'onBack');
+    await modalComponent.onBack();
+    expect(modalComponent.onBack).toHaveBeenCalled();
+  });
```
Comment (matthew2564) [inline src/app/pages/confirm-test-details/components/confirm-submit-modal/__tests__/submit-modal.spec.ts]: This test is a false positive, you're spying on a method, calling it via the test and then checking it has been called

Should be spying on ```spyOn(modalController, 'dismiss');```

and expectation should be 

``` expect(modalController.dismiss).toHaveBeenCalled();```

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/__tests__/submit-modal.spec.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        provideMockStore({}),
+      ],
+    });
+  });
+  beforeEach(waitForAsync(() => {
+    const mockValue = mockFile.mockSlotDetail();
+    spyOn(navMock, 'get').and.returnValue(mockValue);
+    modalFixture = TestBed.createComponent(ConfirmSubmitModal);
+    modalComponent = modalFixture.componentInstance;
+    // spyOn(modalComponent.store$, 'dispatch').and.callFake(() => {});
```
Comment (matthew2564) [inline src/app/pages/confirm-test-details/components/confirm-submit-modal/__tests__/submit-modal.spec.ts]: please remove commented out code

**Review by andrewsetterfield — APPROVED** [review]

**Review by andrewsetterfield — CHANGES_REQUESTED** [review]
can u correct a few linting errors in confirm-test-details.page.ts

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/__mocks__/nav-params.mock.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export class NavParamsMock {
```
Comment (matthew2564) [inline src/app/pages/confirm-test-details/components/confirm-submit-modal/__mocks__/nav-params.mock.ts]: Can we move this to the ```mock/ionic-mocks``` folder which is at the same level as the src file, this mock might be useful elsewhere

File: src/app/pages/confirm-test-details/confirm-test-details.page.ts [hunk 4]
```diff
@@ -217,27 +223,23 @@ export class ConfirmTestDetailsPage extends PracticeableBasePageComponent {
   }
 
   async showConfirmTestDetailsModal(): Promise<void> {
```
Comment (andrewsetterfield) [inline src/app/pages/confirm-test-details/confirm-test-details.page.ts:225]: can you add a unit test for this...i'll send u an example

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/confirm-submit-modal.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  @Input('category') category;
+  @Input('candidateName') candidateName;
+
+  constructor(
+    public modalController: ModalController,
+  ) {
+  }
+
+  ngOnInit(): void {
+  }
```
Comment (matthew2564) [inline src/app/pages/confirm-test-details/components/confirm-submit-modal/confirm-submit-modal.ts]: Can we remove this? Doesn't appear to be used

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/confirm-submit-modal.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  selector: 'confirm-submit-modal',
+  templateUrl: './confirm-submit-modal.html',
+  styleUrls: ['./confirm-submit-modal.scss'],
+})
+export class ConfirmSubmitModal {
+
+  onTestDetailsConfirm: Function;
+  @Input('testOutcome') testOutcome;
+  @Input('category') category;
+  @Input('candidateName') candidateName;
```
Comment (matthew2564) [inline src/app/pages/confirm-test-details/components/confirm-submit-modal/confirm-submit-modal.ts]: Are we able to add some appropriate types to these inputs?

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/__mocks__/confirm-submit-modal.mock.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { DateTime } from '@shared/helpers/date-time';
+import { SlotDetail } from '@dvsa/mes-journal-schema';
+
+export class JournalEarlyStartModalMock {
```
Comment (matthew2564) [inline src/app/pages/confirm-test-details/components/confirm-submit-modal/__mocks__/confirm-submit-modal.mock.ts]: A very similar mock of this already exists here ```src/app/pages/journal/components/journal-early-start-modal/__mocks__/journal-early-start-modal.mock.ts```

Can we just re-use this mock?

File: src/app/pages/confirm-test-details/components/confirm-submit-modal/confirm-submit-modal.ts (hunk 6 above)
Comment (andrewsetterfield) [reply src/app/pages/confirm-test-details/components/confirm-submit-modal/confirm-submit-modal.ts]: Not sure the inputs are nessaccery here.  this seems to work fine.  However agree that types are needed.
image

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**